nms/app/packages/magmalte/app/components/context/RefreshContext.js

```diff
@@ -0,0 +1,178 @@
+ * limitations under the License.
+ *
+ * @flow strict-local
+ * @format
+ */
+'use strict';
+import {FetchEnodebs, FetchGateways} from '../../state/lte/EquipmentState';
+import {
+  FetchSubscriberState,
+  FetchSubscribers,
+} from '../../state/lte/SubscriberState';
+import {useContext, useEffect, useState} from 'react';
+
+export const REFRESH_INTERVAL = 30000;
+
+type Props = {
+  context: typeof React.Context,
+  networkId: string,
+  type: refreshType,
+  interval?: number,
+  id?: string,
+  enqueueSnackbar?: (msg: string, cfg: {}) => ?(string | number),
+  refresh: boolean,
+  lastRefreshTime?: string,
+};
+
+type refreshType = 'subscriber' | 'gateway' | 'enodeb';
+
+export function useRefreshingContext(props: Props) {
+  const ctx = useContext(props.context);
+  const [state, setState] = useState(
+    props.type === 'subscriber'
+      ? {state: ctx.state, sessionState: ctx.sessionState}
+      : ctx.state,
+  );
+
+  const [autoRefreshTime, setAutoRefreshTime] = useState(props.lastRefreshTime);
+
+  async function fetchState(props: FetchProps) {
+    const newState = await fetchRefreshState({
+      type: props.type,
+      networkId: props.networkId,
+      id: props.id,
+      enqueueSnackbar: props.enqueueSnackbar,
+    });
+    if (newState) {
+      setState(() => {
+        if (props.type === 'subscriber') {
+          return {
+            sessionState: newState?.sessionState || {},
+            state: newState?.state || {},
+          };
+        } else {
+          return newState;
+        }
+      });
+    }
+  }
+
+  function updateContext(id?: string, state) {
+    let newState = state;
+    if (id !== null && id !== undefined) {
+      if (props.type === 'subscriber') {
+        newState = {
+          state: {
+            ...ctx.state,
+            // $FlowIgnore
+            [id]: state.state?.[id],
+          },
+          // $FlowIgnore
+          sessionState: Object.keys(state.sessionState || {}).length
+            ? {
+                ...ctx.sessionState,
+                // $FlowIgnore
+                [id]: state.sessionState?.[id],
+              }
+            : {},
+        };
+      } else if (props.type === 'enodeb') {
+        // $FlowIgnore
+        newState = {...ctx.state, [id]: state.enbInfo?.[id]};
+      } else {
+        newState = {...ctx.state, [id]: state?.[id]};
+      }
+    }
+    return ctx.setState(null, null, newState);
+  }
+
+  useEffect(() => {
+    const intervalId = setInterval(
+      () => setAutoRefreshTime(new Date().toLocaleString()),
+      props.interval,
+    );
+    if (!props.refresh) {
+      return clearInterval(intervalId);
+    }
+    return () => {
+      if (props.id !== undefined && props.id !== null) {
+      }
+      updateContext(props.id, state);
+      clearInterval(intervalId);
+    };
+    // eslint-disable-next-line
+  }, [props.interval, props.refresh, state]);
+
+  useEffect(() => {
+    fetchState({
```

Comment:
perhaps we can fetchState only when lastRefreshTime != autoRefreshTime and we can remove dependency of props.refresh from this useEffect